### PR TITLE
Consolidate release notes for 5.0.0 GA

### PR DESCRIPTION
**Pull request by suyograo**

### CODE REVIEW

File: docs/static/releasenotes.asciidoc [hunk 1]
```diff
@@ -1,73 +1,56 @@
...
-* <<alpha1,Logstash 5.0.0-alpha1>>
-
-[[RC1]]
-=== Logstash 5.0.0-rc1 Release Notes
-
-* Windows: Fixed signal HUP warnings on startup ({lsissue}5239[Issue 5239]).
-* Windows: Enabled logging on Windows by fixing log4j config location ({lsissue}5971[Issue 5971]).
-* Configuration errors are now logged to file, not printed to stderr ({lsissue}5975[Issue 5975]).
-* Correctly calculate pipeline `duration_in_millis` metric ({lsissue}5952[Issue 5952]).
+This section summarizes the changes in {version} release.
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: I wouldn't use a variable here because you'll want different versions when you add release notes for subsequent point releases.

File: docs/static/releasenotes.asciidoc [hunk 2]
```diff
@@ -1,73 +1,56 @@
...
+collects various kinds of operational metrics while Logstash processes your data, and all of this information 
+can be queried using simple APIs. Please refer <<monitoring>> for details.
+* Improved throughput performance across the board (up by 2x in some configs) by implementing Event 
+representation in Java. Event is the main object that encapsulates data as it flows through 
+Logstash and provides APIs for the plugins to perform processing. This change also enables 
+faster serialization for future persistence work ({lsissue}4191[Issue 4191]).
+* Breaking Change: Renamed filenames for Debian and RPM artifacts to match Elasticsearch's naming scheme. The metadata is 
+still the same, so upgrades will not be affected. Please refer to the new <<deb-layout, directory structure>>. If you have 
+automated downloads for Logstash, please make sure you use the updated URLs ({lsissue}5100[Issue 5100]).
+* Breaking Change: Introduced a new way to configure application settings for Logstash through a `logstash.yml` file. This
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: A new feature (settings file) shouldn't be listed as a breaking change. There was no prior feature that this new feature removed.

Possibly to note for a breaking change is the fact that command line flags were renamed to match with yaml settings (--foo.bar, etc).

File: docs/static/releasenotes.asciidoc [hunk 3]
```diff
@@ -1,73 +1,56 @@
...
-* Passwords in Logstash logs partially visible when connection to Elasticsearch fails (https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/482[Issue 482]).
+==== General
 
-[[beta1]]
-=== Logstash 5.0-beta1 Release Notes
-
-* Migrated Logstash's internal logging framework to Log4j2. This enhancement provides the following features:
+* A new monitoring feature provides runtime visibility into the Logstash pipeline and its plugins. This component 
+collects various kinds of operational metrics while Logstash processes your data, and all of this information 
+can be queried using simple APIs. Please refer <<monitoring>> for details.
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: Please refer to...

File: docs/static/releasenotes.asciidoc [hunk 4]
```diff
@@ -1,73 +1,56 @@
...
+* Breaking Change: Introduced a new way to configure application settings for Logstash through a `logstash.yml` file. This
+file is typically located in `LS_HOME/config` or `/etc/logstash` when installed via packages. Logstash will
+not be able to start without this file, so please make sure to pass in `path.settings` if you are starting
+Logstash manually after installing it via a package (RPM or DEB) ({lsissue}4401[Issue 4401]).
+* Breaking Change: Most of the long form CLI options have been renamed to match the settings defined in `logstash.yml`
+* Breaking Change: For plugin developers, the Event class has a new API to access its data. You will no longer be able to
+directly use the Event class through the Ruby hash paradigm. All the plugins packaged with Logstash have
+been updated to use the new API, and their versions have been bumped to the next major. Please refer <<event-api>> for details.({lsissue}5141[Issue 5141])
+* Breaking Change: Environment variables inside the Logstash config are evaluated by default. There is no need to specify 
+`--allow-en` feature flag.
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: typo, `--allow-env`

File: docs/static/releasenotes.asciidoc [hunk 5]
```diff
@@ -1,73 +1,56 @@
...
+* Breaking Change: Renamed `bin/plugin` to `bin/logstash-plugin`. This is to prevent `PATH` being polluted when other 
+components of the Elastic stack are installed on the same instance ({lsissue}4891[Issue 4891]).
+* Added support for `DEBUG=1` when running any plugin-related commands. This option gives you a bit more information
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: Maybe rephrase this as "This option gives more information that is useful when debugging unexpected behavior in bin/logstash-plugin"

File: docs/static/releasenotes.asciidoc [hunk 6]
```diff
@@ -1,73 +1,56 @@
...
+Logstash and provides APIs for the plugins to perform processing. This change also enables 
+faster serialization for future persistence work ({lsissue}4191[Issue 4191]).
+* Breaking Change: Renamed filenames for Debian and RPM artifacts to match Elasticsearch's naming scheme. The metadata is 
+still the same, so upgrades will not be affected. Please refer to the new <<deb-layout, directory structure>>. If you have 
+automated downloads for Logstash, please make sure you use the updated URLs ({lsissue}5100[Issue 5100]).
+* Breaking Change: Introduced a new way to configure application settings for Logstash through a `logstash.yml` file. This
+file is typically located in `LS_HOME/config` or `/etc/logstash` when installed via packages. Logstash will
+not be able to start without this file, so please make sure to pass in `path.settings` if you are starting
+Logstash manually after installing it via a package (RPM or DEB) ({lsissue}4401[Issue 4401]).
+* Breaking Change: Most of the long form CLI options have been renamed to match the settings defined in `logstash.yml`
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: missing period: `logstash.yml`.

File: docs/static/releasenotes.asciidoc [hunk 7]
```diff
@@ -1,73 +1,56 @@
...
+directly use the Event class through the Ruby hash paradigm. All the plugins packaged with Logstash have
+been updated to use the new API, and their versions have been bumped to the next major. Please refer <<event-api>> for details.({lsissue}5141[Issue 5141])
+* Breaking Change: Environment variables inside the Logstash config are evaluated by default. There is no need to specify 
+`--allow-en` feature flag.
+* Breaking Change: Renamed `bin/plugin` to `bin/logstash-plugin`. This is to prevent `PATH` being polluted when other 
+components of the Elastic stack are installed on the same instance ({lsissue}4891[Issue 4891]).
+* Added support for `DEBUG=1` when running any plugin-related commands. This option gives you a bit more information 
+about what the bundler is doing.
+* Logging Changes: Migrated Logstash's internal logging framework to Log4j2. This enhancement provides the following features:
 ** Support for changing the <<logging>> level dynamically at runtime through REST endpoints. New APIs have been exposed
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: This point mentions _node/logging twice in two different sentences that seem to mean the same thing?

File: docs/static/releasenotes.asciidoc [hunk 8]
```diff
@@ -1,73 +1,56 @@
...
+still the same, so upgrades will not be affected. Please refer to the new <<deb-layout, directory structure>>. If you have 
+automated downloads for Logstash, please make sure you use the updated URLs ({lsissue}5100[Issue 5100]).
+* Breaking Change: Introduced a new way to configure application settings for Logstash through a `logstash.yml` file. This
+file is typically located in `LS_HOME/config` or `/etc/logstash` when installed via packages. Logstash will
+not be able to start without this file, so please make sure to pass in `path.settings` if you are starting
+Logstash manually after installing it via a package (RPM or DEB) ({lsissue}4401[Issue 4401]).
+* Breaking Change: Most of the long form CLI options have been renamed to match the settings defined in `logstash.yml`
+* Breaking Change: For plugin developers, the Event class has a new API to access its data. You will no longer be able to
+directly use the Event class through the Ruby hash paradigm. All the plugins packaged with Logstash have
+been updated to use the new API, and their versions have been bumped to the next major. Please refer <<event-api>> for details.({lsissue}5141[Issue 5141])
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: Please refer to...

File: docs/static/releasenotes.asciidoc [hunk 9]
```diff
@@ -1,73 +1,56 @@
...
+components of the Elastic stack are installed on the same instance ({lsissue}4891[Issue 4891]).
+* Added support for `DEBUG=1` when running any plugin-related commands. This option gives you a bit more information 
+about what the bundler is doing.
+* Logging Changes: Migrated Logstash's internal logging framework to Log4j2. This enhancement provides the following features:
 ** Support for changing the <<logging>> level dynamically at runtime through REST endpoints. New APIs have been exposed
 under `_node/logging` to update log levels. Also a new endpoint `_node/logging` was added to return all existing loggers.
 ** Configurable file rotation policy for logs. The default is per-day.
 ** Support for component-level or plugin-level log settings.
 ** Unified logging across Logstash's Java and Ruby code.
 ** Logs are now placed in the `LS_HOME/logs` directory, which is configurable via the `path.logs` setting.
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: THis depends on the package being installed. It's /var/log/logstash/... for rpm/deb by default.

File: docs/static/releasenotes.asciidoc (hunk 8 above)
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: There is a missing space and the period appears in the wrong place. Should say: ...details ({lsissue}5141[Issue 5141]).

File: docs/static/releasenotes.asciidoc [hunk 10]
```diff
@@ -1,73 +1,56 @@
...
+automated downloads for Logstash, please make sure you use the updated URLs ({lsissue}5100[Issue 5100]).
+* Breaking Change: Introduced a new way to configure application settings for Logstash through a `logstash.yml` file. This
+file is typically located in `LS_HOME/config` or `/etc/logstash` when installed via packages. Logstash will
+not be able to start without this file, so please make sure to pass in `path.settings` if you are starting
+Logstash manually after installing it via a package (RPM or DEB) ({lsissue}4401[Issue 4401]).
+* Breaking Change: Most of the long form CLI options have been renamed to match the settings defined in `logstash.yml`
+* Breaking Change: For plugin developers, the Event class has a new API to access its data. You will no longer be able to
+directly use the Event class through the Ruby hash paradigm. All the plugins packaged with Logstash have
+been updated to use the new API, and their versions have been bumped to the next major. Please refer <<event-api>> for details.({lsissue}5141[Issue 5141])
+* Breaking Change: Environment variables inside the Logstash config are evaluated by default. There is no need to specify
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: Should say, "There is no need to specify the `--allow-en` feature flag."

File: docs/static/releasenotes.asciidoc [hunk 11]
```diff
@@ -1,73 +1,56 @@
...
-* Breaking Change: The index template for 5.0 has been changed to reflect Elasticsearch's mapping changes. Most importantly,
-the subfield for string multi-fields has changed from `.raw` to `.keyword` to match Elasticsearch's default behavior
-(https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/386[Issue 386]). See <<breaking-changes>> for details about how this change affects new and existing users.
-* Added `check_connection_timeout` parameter, which has a default of 10m.
-
-[[alpha5]]
-=== Logstash 5.0-alpha5 Release Notes
-
+** Changed the default log severity level to `INFO` instead of `WARN` to match Elasticsearch.
+** Logstash can now emit its log in structured, JSON format. Specify `--log.format=json` in the settings file
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: This says "in the settings file" but the example is a flag `--log.format`. Maybe remove the `--` part and make it valid yaml?

File: docs/static/releasenotes.asciidoc [hunk 12]
```diff
@@ -1,73 +1,56 @@
...
+or via the command line ({lsissue}1569[Issue 1569])
+* Added support for systemd so you can now manage Logstash as a service on most Linux distributions ({lsissue}5012[Issue 5012]).
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: systemd and upstart

File: docs/static/releasenotes.asciidoc [hunk 13]
```diff
@@ -1,73 +1,56 @@
...
+not be able to start without this file, so please make sure to pass in `path.settings` if you are starting
+Logstash manually after installing it via a package (RPM or DEB) ({lsissue}4401[Issue 4401]).
+* Breaking Change: Most of the long form CLI options have been renamed to match the settings defined in `logstash.yml`
+* Breaking Change: For plugin developers, the Event class has a new API to access its data. You will no longer be able to
+directly use the Event class through the Ruby hash paradigm. All the plugins packaged with Logstash have
+been updated to use the new API, and their versions have been bumped to the next major. Please refer <<event-api>> for details.({lsissue}5141[Issue 5141])
+* Breaking Change: Environment variables inside the Logstash config are evaluated by default. There is no need to specify 
+`--allow-en` feature flag.
+* Breaking Change: Renamed `bin/plugin` to `bin/logstash-plugin`. This is to prevent `PATH` being polluted when other 
+components of the Elastic stack are installed on the same instance ({lsissue}4891[Issue 4891]).
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: The company style guide now says that we should capitalize Elastic Stack. I'll try to get around to fixing this across the doc, but a very low priority.

File: docs/static/releasenotes.asciidoc [hunk 14]
```diff
@@ -1,73 +1,56 @@
...
-* Correctly calculate pipeline `duration_in_millis` metric ({lsissue}5952[Issue 5952]).
+This section summarizes the changes in {version} release.
 
 [float]
-==== Output Plugins
-
-*`Elasticsearch`*:
-
-* Passwords in Logstash logs partially visible when connection to Elasticsearch fails (https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/482[Issue 482]).
+==== General
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: I wonder if it's worth organizing these into sub-lists by category (like we already do for logging changes): for example, packaging changes, CLI changes, plugin-related changes, etc. Maybe this is something I can do later after all high priority work is done?

File: docs/static/releasenotes.asciidoc [hunk 15]
```diff
@@ -234,122 +130,30 @@ for content in XML elements. This option allows you to force text content and at
...
 
+* Breaking Change: The index template for 5.0 has been changed to reflect Elasticsearch's mapping changes. Most importantly,
+the subfield for string multi-fields has changed from `.raw` to `.keyword` to match Elasticsearch's default behavior
+(https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/386[Issue 386]). See <<breaking-changes>> for details about how this change affects new and existing users.
+* Added `check_connection_timeout` parameter, which has a default of 10m.
+* Added the ability for the plugin to choose which default template to use based on the Elasticsearch version (https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/401[Issue 401]).
+* Elasticserach output is now fully threadsafe. This means internal resources can be shared among multiple
+`output { elasticsearch {} }` instances.
+* Added sniffing improvements so any current connections don't have to be closed/reopened after a sniff round.
+* Introduced a connection pool to efficiently reuse connections to Elasticsearch backends.
```
Comment (jordansissel) [inline docs/static/releasenotes.asciidoc]: I think we can do without the 'efficiently' word. pooling and connection reuse *are* efficiency improvements :)

File: docs/static/releasenotes.asciidoc [hunk 16]
```diff
@@ -1,73 +1,56 @@
...
-the subfield for string multi-fields has changed from `.raw` to `.keyword` to match Elasticsearch's default behavior
-(https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/386[Issue 386]). See <<breaking-changes>> for details about how this change affects new and existing users.
-* Added `check_connection_timeout` parameter, which has a default of 10m.
-
-[[alpha5]]
-=== Logstash 5.0-alpha5 Release Notes
-
+** Changed the default log severity level to `INFO` instead of `WARN` to match Elasticsearch.
+** Logstash can now emit its log in structured, JSON format. Specify `--log.format=json` in the settings file
+or via the command line ({lsissue}1569[Issue 1569])
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: missing end period: [Issue 1569]).

**Comment by jordansissel** [comment]
General comment: Should we put all breaking changes into a separate section? @dedemorton what do you think? My aim is to improve discoverability of breaking changes.

File: docs/static/releasenotes.asciidoc [hunk 17]
```diff
@@ -78,152 +61,65 @@ to filter the API response ({lsissue}5609[Issue 5609]).
...
-* Made this output a shareable instance across multiple pipeline workers. This ensures efficient use of resources like broker
-TCP connections, internal producer buffers, and so on.
-
-*`Tcp`*:
-
-* Added SSL/TLS support for certificate-based encryption.
-
+* Added support for Kafka broker 0.10. This plugin now supports SSL based encryption. This release 
+changed a lot of configuration, so it is not backward compatible. Also, this version will not work 
+with older Kafka brokers
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: Missing end period: Kafka brokers.

File: docs/static/releasenotes.asciidoc [hunk 18]
```diff
@@ -1,73 +1,56 @@
 [[releasenotes]]
 == Release Notes
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: Shouldn't we also link to the breaking changes doc?

**Comment by suyograo** [comment]
@jordansissel @dedemorton updated

* Added a link to breaking changes
* Fixed typos and other comments

File: docs/static/releasenotes.asciidoc [hunk 19]
```diff
@@ -1,46 +1,49 @@
...
 ** Support for component-level or plugin-level log settings.
 ** Unified logging across Logstash's Java and Ruby code.
-** Logs are now placed in the `LS_HOME/logs` directory, which is configurable via the `path.logs` setting.
+** Logs are now placed in the `LS_HOME/logs` directory, which is configurable via the `path.logs` setting. For deb/rpm packages 
+logs are placed in `/var/log/logstash/` by default.
 ** Changed the default log severity level to `INFO` instead of `WARN` to match Elasticsearch.
-** Logstash can now emit its log in structured, JSON format. Specify `--log.format=json` in the settings file
-or via the command line ({lsissue}1569[Issue 1569])
-* Added support for systemd so you can now manage Logstash as a service on most Linux distributions ({lsissue}5012[Issue 5012]).
+** Logstash can now emit its log in structured, JSON format. Specify log.format=json` in the settings file
```
Comment (dedemorton) [inline docs/static/releasenotes.asciidoc]: you have an unnecessary back tick in log.format=json`

**Comment by dedemorton** [comment]
@suyograo noted a typo, otherwise LGTM

**Comment by jordansissel** [comment]
I haven't checked the rendering, but the content LGTM.
